src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -165,6 +155,54 @@ public async Task<WebSocket> AcceptAsync(WebSocketAcceptContext acceptContext)
 
                 return WebSocket.CreateFromStream(opaqueTransport, isServer: true, subProtocol: subProtocol, keepAliveInterval: keepAliveInterval);
             }
+
+            public static bool CheckSupportedWebSocketRequest(string method, IHeaderDictionary requestHeaders)
+            {
+                if (!string.Equals("GET", method, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+
+                foreach (var pair in requestHeaders)
+                {
+                    if(!requestHeaders[HeaderNames.Connection].Contains(HeaderNames.Upgrade))
+                    {
+                        return false;
+                    }
+
+                    if(!requestHeaders[HeaderNames.Upgrade].Contains(Constants.Headers.UpgradeWebSocket))
+                    {
+                        return false;
+                    }
+
+                    if(!requestHeaders[HeaderNames.SecWebSocketVersion].Contains(Constants.Headers.SupportedVersion))
+                    {
+                        return false;
+                    }
+
+                    if(requestHeaders.ContainsKey(HeaderNames.SecWebSocketKey))
+                    {
+                        if(!HandshakeHelpers.IsRequestKeyValid(pair.Value))
+                            return false;
+                    }
+                }
+
+                // WebSockets are long lived; so if the header values are valid we switch them out for the interned versions.
+                if (requestHeaders[HeaderNames.Connection].Count == 1)
```

Comment:
This is also broken without `GetCommaSeparatedValues`. You could replace "Upgrade, Something" with "Upgrade" which is bad.

Reply:
Same here as well!

Reply:
Maybe we need a test for this?